Shared/DeferredUtils.swift

```diff
@@ -99,9 +99,13 @@ public func accumulate<T>(_ thunks: [() -> Deferred<Maybe<T>>]) -> Deferred<Mayb
     var onValue: ((T) -> Void)!
     var onResult: ((Maybe<T>) -> Void)!
 
+    // onValue and onResult both hold refrences to each other. 
+    // niling them out before exiting breaks a refrence cycle
```

Comment:
"refrence" => "reference", on both lines. Closing period.

Worth noting that we can't use `unowned` here 'cos thunks aren't class types.